scripts/zulip_emoji_merge_delegate.py

```diff
@@ -25,74 +25,70 @@
-    print(f"matched: '{message}'")
     # Check for emoji reactions
     reactions = message['reactions']
     has_peace_sign = any(reaction['emoji_name'] == 'peace_sign' for reaction in reactions)
     has_bors = any(reaction['emoji_name'] == 'bors' for reaction in reactions)
     has_merge = any(reaction['emoji_name'] == 'merge' for reaction in reactions)
+    match = pr_pattern.search(content)
+    if match:
+        print(f"matched: '{message}'")
 
-    pr_url = f"https://api.github.com/repos/leanprover-community/mathlib4/pulls/{PR_NUMBER}"
-
-    print('Removing peace_sign')
-    result = client.remove_reaction({
-        "message_id": message['id'],
-        "emoji_name": "peace_sign"
-    })
-    print(f"result: '{result}'")
-    print('Removing bors')
-    result = client.remove_reaction({
-        "message_id": message['id'],
-        "emoji_name": "bors"
-    })
-    print(f"result: '{result}'")
-
-    print('Removing merge')
-    result = client.remove_reaction({
-        "message_id": message['id'],
-        "emoji_name": "merge"
-    })
-    print(f"result: '{result}'")
-
-    if 'delegated' == LABEL:
-        print('adding delegated')
-
-        client.add_reaction({
-            "message_id": message['id'],
-            "emoji_name": "peace_sign"
-        })
-    elif 'ready-to-merge' == LABEL:
-        print('adding ready-to-merge')
+        # removing previous emoji reactions
+        print("Removing previous reactions, if present.")
         if has_peace_sign:
-            client.remove_reaction({
+            print('Removing peace_sign')
+            result = client.remove_reaction({
                 "message_id": message['id'],
                 "emoji_name": "peace_sign"
             })
-        client.add_reaction({
-            "message_id": message['id'],
-            "emoji_name": "bors"
-        })
-    elif LABEL.startswith("[Merged by Bors]"):
-        print('adding [Merged by Bors]')
-        if has_peace_sign:
-            client.remove_reaction({
+            print(f"result: '{result}'")
+        if has_bors:
+            print('Removing bors')
+            result = client.remove_reaction({
+                "message_id": message['id'],
+                "emoji_name": "bors"
+            })
+            print(f"result: '{result}'")
+        if has_merge:
+            print('Removing merge')
+            result = client.remove_reaction({
+                "message_id": message['id'],
+                "emoji_name": "merge"
+            })
+            print(f"result: '{result}'")
+
+        # applying appropriate emoji reaction
+        print("Applying reactions, as appropriate.")
+        if 'delegated' == LABEL:
+            print('adding delegated')
+            client.add_reaction({
                 "message_id": message['id'],
                 "emoji_name": "peace_sign"
             })
-        if has_bors:
-            client.remove_reaction({
+        elif 'ready-to-merge' == LABEL:
```

Comment:
Right now it looks like if a PR is both delegated and merged, only the delegated emoji will be applied, right?

Perhaps it would be good to change the order of the ifs so that the "merge" emoji takes priority over "bors", which takes priority over "peace_sign"?

Reply:
I swapped the two, although I think that the action will only look at the first `bors x` that is applied.